read_buffer/src/chunk.rs

```diff
@@ -236,10 +237,26 @@ impl Chunk {
     /// added to the `Table`. Otherwise a new `Table` with a single `RowGroup`
     /// will be created.
     pub fn upsert_table(&self, table_name: impl Into<String>, table_data: RecordBatch) {
+        // Approximate heap size of record batch.
+        let rb_size = table_data
+            .columns()
+            .iter()
+            .map(|c| c.get_buffer_memory_size())
+            .sum::<usize>();
+
         // This call is expensive. Complete it before locking.
         let row_group = RowGroup::from(table_data);
         let table_name = table_name.into();
 
+        let rg_size = row_group.size();
+        info!(
+            "RB row group added: rb_size {:?} rg_size {:?} ({:?}%)\n{}",
+            rb_size,
+            rg_size,
+            rg_size as f64 / rb_size as f64,
+            &row_group,
+        );
```

Comment:
If you want, you can emit `rb_size` etc as structured fields (and get the `rg_size=...` for free with something like 

```suggestion
        let rg_size = row_group.size();
        let percent =  rg_size as f64 / rb_size as f64;
        info!(
            rb_size, rg_size, percent,
            ?row_group,
            "RB row group added",
        );
```

I also don't really sure how helpfl the

Reply:
+1

Keeping the log messages fairly static but having the data as structured fields is super invaluable for retrospective debugging via log analysis - it saved my skin many times after a customer dumped a load of logs on us post-incident.